mlir/include/mlir/Dialect/AMX/AMX.td

```diff
@@ -88,14 +151,14 @@ def TileZeroOp : AMX_Op<"tile_zero", [Pure]> {
     Example:
 
     ```mlir
-      %0 = amx.tile_zero : vector<16x16xbf16>
+      %0 = amx.tile_zero : <16x16xbf16>
     ```
   }];
   let results = (outs
-    VectorOfRankAndType<[2], [F32, BF16, I32, I8]>:$res);
+    AMX2DTileOf<[F32, F16, BF16, I32, I8]>:$res);
```

Comment:
`AMX2DTileOf` seems to always allow only a few sets of element types. Perhaps lift that to the type's declaration to avoid repetition? Like `AMXTileAll` and `AMXTileF16` (and assume they're all 2D)?

Reply:
I added more specific constrains and all tiles are 2D now.